### ISSUE TITLE
대성고 수집 데이터 활용 object detection

### ISSUE THREAD

**Issue by JNY03** [HBNU-SWUNIV/come-capstone26-physicalAI#293]
대성고 실습에서 go1으로 촬영한 데이터로 object detection + depth 모델 구축

**Comment by JNY03** [comment]
## 데이터 전처리 진행 전 raw data 분석

[dataset_analysis_for_training.md](https://github.com/user-attachments/files/26432759/dataset_analysis_for_training.md)

**Comment by JNY03** [comment]
## 이미지 라벨링 툴로 CVAT 활용
학생들이 사용할 것을 고려해 보안성, 사용 난이도, 활용 범위 등 기준에 적합한 툴 선정

**Comment by JNY03** [comment]
## yolo world 도메인 학습 진행
- yolo-world 기존 모델 적용 결과 성능이 낮고
- grounding dino 기존 모델 적용 결과 속도가 느린 문제
**-> 라벨링한 데이터로 도메인 학습 진행하기로 결정**
- 학습 파이프라인 경로: /home/jny/Go1_OD/yolo_world_domain_training

## 학습 내용, 결과 요약
- 대상 데이터: go1_대성고 이미지 라벨링 데이터
- 실험 범위: YOLO-World direct fine-tuning
- 사용 모델: YOLOWorld (`yolov8s-worldv2.pt` 기반)
- 최종 best.pt 경로: `direct_finetune/runs/train/train5/weights/best.pt`
- 학습 전후 test 결과 비교(대성고 이미지 test set):
```

구분	Precision	Recall	mAP50	mAP50-95
학습 전 pretrained	0.979	0.129	0.200	0.144
학습 후 fine-tuned	0.860	0.862	0.892	0.539
변화량:

Precision: -0.119
Recall: +0.734
mAP50: +0.692
mAP50-95: +0.395

```
다음은 연구실 촬영 데이터를 넣고 학습 전후 성능 비교해볼 계획

**Comment by JNY03** [comment]
## pretrained 모델과 fintuned모델 unseen data 시각화 비교

### pretrained
<img width="464" height="400" alt="Image" src="https://github.com/user-attachments/assets/599a21ab-bf0c-4d76-8ec0-583085852a4f" />

<img width="464" height="400" alt="Image" src="https://github.com/user-attachments/assets/f7529d20-c650-40b1-902d-7a4e43c5d53b" />

### finetuned

<img width="464" height="400" alt="Image" src="https://github.com/user-attachments/assets/f78d7ce9-73f5-408e-88c6-cc2277fdfb80" />

<img width="464" height="400" alt="Image" src="https://github.com/user-attachments/assets/e34cc135-fe9e-4823-b6d4-4ed063c53c66" />

**Comment by khw18033** [comment]
## 현재 정책표 및 우선순위 정리

**정책표 (Policy):**

| Group | Action | 동작 |
|-------|--------|------|
| **AGENT** | stop, 4.0초 | 긴급 정지 4초 |
| **VEHICLE** | stop, 4.0초 | 긴급 정지 4초 |
| **HARD_OBSTACLE** | avoid | Bounding box 기반 좌/우 반대방향 회피 |
| **LOW_OBSTACLE** | observe | 인식만 (특정 동작 없음) |
| **THIN_OBSTACLE** | stop_then_back | 긴급 정지 1초 → 뒤로 이동 0.5초 |
| **GROUND_HAZARD** | stop_then_back | 긴급 정지 2초 → 뒤로 이동 0.5초 |
| **UNKNOWN_OBSTACLE** | stop, 2.0초 | 긴급 정지 2초 |

**우선순위 (Priority - 낮을수록 먼저 처리):**

| 순위 | Group | 우선순위 값 |
|------|-------|----------|
| 1 | AGENT | 0 |
| 2 | VEHICLE | 1 |
| 3 | GROUND_HAZARD | 2 |
| 4 | HARD_OBSTACLE | 3 |
| 5 | THIN_OBSTACLE | 4 |
| 6 | UNKNOWN_OBSTACLE | 5 |
| 7 | LOW_OBSTACLE | 6 |

GROUP 매핑이 안되었거나 인식만 되고 class 분류가 안 된 장애물은 UNKNOWN_OBSTATLE로 분류

**Comment by JNY03** [comment]
## 해결할 문제 
- **처리 속도 느림**: 한번에 3가지 모델 처리(worker 1->3)
- **AGENT 편향**: OD 모델 내부 우선순위 설정으로 인해 발생(우선순위 설정 완화)
- **안정성 부족**: 한 객체에 관한 판단이 수시로 바뀌는 문제(tracking 적용)
